lib/rich_text.rb

```diff
@@ -118,16 +118,36 @@ def sanitize(text)
 
     def linkify(text, mode = :urls)
       link_attr = 'rel="nofollow noopener noreferrer" dir="auto"'
-      Rinku.auto_link(ERB::Util.html_escape(text), mode, link_attr) do |url|
-        url = shorten_host(url, Settings.linkify_hosts, Settings.linkify_hosts_replacement)
-        shorten_host(url, Settings.linkify_wiki_hosts, Settings.linkify_wiki_hosts_replacement) do |path|
-          path.sub(Regexp.new(Settings.linkify_wiki_optional_path_prefix || ""), "")
-        end
+      Rinku.auto_link(unshorten_links(ERB::Util.html_escape(text)), mode, link_attr) do |url|
+        shorten_link(url)
       end.html_safe
     end
 
     private
 
+    def unshorten_links(text)
+      Settings.linkify.paths.each do |path_rule|
```

Comment:
Probably worth providing for the case where this setting is not present. This appears to be done for `Settings.linkify_hosts`, as `shorten_host` does `hosts&.include(host)`.

```suggestion
      linkify_paths = Array.wrap(Settings.linkify&.paths)
      linkify_paths.each do |path_rule|
```